infrasec/tf/atlantis.md

```diff
@@ -1,14 +1,219 @@
+After you add the deploy key to GitHub, you should get an automated email to confirm the key was added.
+
+### Step 1.5: Store the Key in AWS SSM
+
+Log into the console for the account you want to deploy Atlantis in (in our case, `transcom-gov-milmove-exp`), and add your key to AWS Systems Manager > Parameter Store with the name `/atlantis-global/atlantis_key` as type `SecureString`, using "My current account" as the KMS key source.
+
+### Step 2: Do You Even Docker
+
+Because we use already use our own base Docker image on MilMove, we're continuing the established pattern to deploy Atlantis. We'll create a docker image for Atlantis and put that image in ECR with all our other images. Next, we can create (and define) an ECS task to retrieve our Atlantis image from ECR. This way, we can require our Atlantis user to operate using our specific desired conditions.
+
+The module creates the task and the role.
+
+Make a dockerfile that runs inside the base circleci image. You'll need to install Terraform and Atlantis. Because our setup requires us to run docker-in-docker, you'll need to run the install as `USER root`, and then switch back to `USER circleci` after the install to be compatible with our base image.
+See [this PR](https://github.com/transcom/circleci-docker/pull/98) for the implementation.
+
+### Add the Atlantis bucket and iam user
+
+Create an `atlantis-global` directory in your desired account:
+`mkdir -p transcom-gov-milmove-exp/atlantis-global`
+
+BUCKET: Add the bucket, version, and provider files following the steps (more or less) done in the bootstrapping document in [the Atlantis section](https://github.com/trussworks/legendary-waddle/blob/master/docs/how-to/bootstrap-new-aws-account.md#atlantis).
+
+IAM role: Add the IAM role. TODO: add note explaining why we do this differently from circleci (and need only roles but not a user). At this point you can go ahead and add the `atlantis.yaml` with a list of directories Atlantis needs to access, and [update the PR](https://github.com/transcom/transcom-infrasec-gov/pull/275) template to include the new directory.
+
+### Add & validate a certificate for Atlantis
+
+TODO: add certificate screenshots from ACM in console
+
+Add a certificate to ACM, which [manages our certificates for us](https://github.com/transcom/transcom-infrasec-gov/pull/279).
+
+You will need to find the `arn` to plug it into the [atlantis module](https://registry.terraform.io/modules/terraform-aws-modules/atlantis/aws/latest) you'll call in the next step. You can find this in the console's Certificate Manager after merging the PR to add the certificate.
+
+![TODO](images/atlantis_acm1.png)
+
+Because of AWS fuckery, we have to validate the cert in a different location. See this PR for [where we're validating the certs we create](https://github.com/transcom/transcom-infrasec-com/pull/1397).
```

Comment:
I actually laughed out loud, but I suggest we change this before it goes live.

Reply:
Of course